### DISCUSSION TITLE
Working with ADVectorVariableValue

### DISCUSSION THREAD

**Discussion by aarograh** [idaholab/moose#22072] [Q&A General] [answered]
I'm working on updating submodules and MOOSE environment for my app.  The update is a small step forward (`moose-libmesh` version updated from `2022.03.18` to `2022.03.28`) but breaks several of my tests with this error:
```
Floating point exception signaled (floating point divide by zero)!

To track this down, compile in debug mode, then in gdb do:
  break libmesh_handleFPE
  run ...
  bt
```

Attaching the debugger and following the instructions given by the error indicates that the error occurs at this line:
```
if (_velocity[_qp].norm() < 1.e-8)
```
`_velocity` is a `ADVectorVariableValue` object.  It seems that in the update, the result of the `.norm()` function call has changed in some way.  So I have 2 questions:
1. I cannot for the life of me determine how to print out the components of `_velocity[_qp]`.  It looks like it's ultimately a templated `libMesh` object.  I figure if I could just inspect the object a bit, I might be able to determine what changed and fix it.
2. Any ideas what may have changed with the norm function during that update?

**Comment by GiudGiud** [comment, marked as answer]
Hello

2. no idea and I doubt it. @lindsayad ?
1. I'm guessing the type here, but likely you can use ``_console << _velocity[_qp](0) << " " << _velocity[_qp](1);``

You probably do not need to compute the norm of an ADReal(Vector) which could be going to 0. At 0, this operation is not differntiable. 

Likely you only need to take the norm of the Real(Vector), so do:
```
_velocity[_qp].value().norm() < 1.e-8
```
or use the normsq() function, avoiding the square root

Guillaume

**Reply by aarograh** [reply]
Ok, the `norm_sq` function looks like it does the trick.  I'm guessing the `.value()` suggestion would also work.  Good to know about both of those!  Thanks!

**Reply by lindsayad** [reply]
More likely is that we added a new `FloatingPointExceptionGuard` in MOOSE that caught floating point exceptions that weren't being caught before

**Reply by GiudGiud** [reply]
want to create the issue or should I do it?

**Reply by GiudGiud** [reply]
ok I created #22076
